### DISCUSSION TITLE
Scikit-build Developer Meeting #6b: 2023-05-05

### DISCUSSION THREAD

**Discussion by jcfr** [scikit-build/scikit-build#970] [Community meeting notes]
## Agenda 

- Debrief of the Python conference by @henryiii.

## Attending

- Matt McCormick
- Henry Scheiner
- Jean-Christophe Fillion-Robin

## Discussion

The meeting discussed the following topics:

- Standardization of the plugin interface for dynamic metadata so that plugins can be implemented in languages such as Rust:
  - https://discuss.python.org/t/proposal-for-dynamic-metadata-plugins/25237
  - https://github.com/scikit-build/scikit-build-core/issues/230
  
- WebAssembly and Python were discussed, and a PyCon 2023 talk about Pyodide was highlighted:
  - Talk: [Pyodide: A Python distribution for the browser](https://us.pycon.org/2023/events/webassembly-summit/)
  
- Support for building Pyodide wheels on cibuildwheel:
  - https://github.com/pypa/cibuildwheel/issues/1002
  - https://github.com/pypa/cibuildwheel/issues/1454
  
- Pybind11 issues related to the of testing of Pyodide were discussed:
  - https://github.com/pybind/pybind11/pull/4642
  - https://github.com/pybind/scikit_build_example/pull/86
  
- The WebAssembly Component Model was discussed:
  - https://www.fermyon.com/blog/webassembly-component-model
  - https://github.com/WebAssembly/component-model

### Todos & Ideas

The following are the todos and ideas mentioned during the meeting:

- Add testing in `pybind11` to check if Pyodide wheels are working.
- Improve support for testing Pyodide wheels in `scikit-build-core`.
- Look into streamlining the creation of new projects (by creating a command-line tool (allowing to run command such as `skbuild new`), or look into templates like GitHub template or cookicutter).
- Revamp the `scikit-build.org` website with a landing page, high-level usage, a list of projects that use `scikit-build-core` (with a format similar to https://cibuildwheel.readthedocs.io/en/stable/working-examples/).
- Create a GitHub repository, `scikit-build.github.io`, using either Sphinx, Jekyll, or Hugo.

### Recording and Transcript

- NA

**Comment by henryiii** [comment]
A few things I meant to ask or mention:

* Should we try to organize SciPy sprints?
* Could we ask about being included in SciPy tools updates?
